pkg/manifests/configaware_test.go

```diff
@@ -235,3 +235,44 @@ spec:
 	assert.NoError(t, err)
 	assert.Equal(t, expectedPatch, string(patch))
 }
+
+const mistakenConf = `
+version: 1
+commandUpdated: # <-- because this is commandUpdated, patchFile is ignored
+  generators:
+    - command: |
+       echo "apiVersion: extensions/v1beta1
+       kind: Deployment
+       metadata:
+         name: helloworld
+       spec:
+         template:
+           metadata:
+             labels:
+              name: helloworld
+           spec:
+             containers:
+             - name: greeter
+               image: quay.io/weaveworks/helloworld:master-a000001
+       ---
+       apiVersion: v1
+       kind: Namespace
+       metadata:
+         name: demo"
+  patchFile: patchfile.yaml
```

Comment:
Ideally, I think we should just reject this file and parse strictly (although returning an error when no updates are done is better and should have is certainly an improvement).

Reply:
Yeah; validating against a schema was my next stop. While the particular error here would change, there would probably still be ways to have a config that doesn't quite work.